libs/single-spa-angular/elements/src/element-props-providers.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable, StaticProvider } from '@angular/core';
+import { ReplaySubject, Subject } from 'rxjs';
+
+@Injectable()
+export class SingleSpaElementPropsService<T> {
```

Comment:
I don't see how this is connected to the original issue of supporting multiple instances of a single-spa-angular element or parcel. I'm not against the feature, but don't think it should be part of this PR

Reply:
It is probably not essential but in some way fits in the whole problem with multiple angular element instances and single-spa.
As I said before, in angular there is a problem, that we have a divide between the properties and the component, that's a big difference between angular and react for example. On the other hand the angular elements are practically singletons - so we should have a Subject for each web component instance and in some way differentiate between the Subjects. So we pass the parcel name as an attribute to the angular component (over the template as a function) and then we read the instance specific properties from the props service....

generally in the component.html we have something like:
`<p [attr.parcel]="parcel" (click)="selected()">works with {{item}}!</p>`
and in the component.ts something along the lines:
```
 @Input()
  public item!: string;

  @Input()
  public parcel!: string;

  public selected = () => {
  };
  private subscription?: Subscription;

  constructor(private _cd: ChangeDetectorRef,
              private propsService: SingleSpaElementPropsService<WrapperParcelProps>) {
  }

  ngOnDestroy(): void {
    if (this.subscription) {
      this.subscription.unsubscribe();
    }
  }

  ngOnInit(): void {
    this.subscription = this.propsService.getProps(this.parcel).subscribe(
      (props: WrapperParcelProps) => {

```

Reply:
If Angular Elements do not support having multiple instances of the same custom element with different properties, then I don't think we should either in single-spa-angular. Doing so seems like it would be a hacky workaround for a limitation within Angular Elements. I don't know much about Angular elements, so if I'm misunderstanding please let me know.

Reply:
The problem is not angular elements per se, the web components could use properties just fine. The problem in fact is the way single-spa-angular is implemented - generally using global subject for the props and passing them in parallel to the creation of the components. In single-spa-react for example, if I understand the code correctly, [getElementToRender](https://github.com/single-spa/single-spa-react/blob/4d77396990aa68284d2dc82a23ff1b3d3a6d6ed6/src/single-spa-react.js#L234) the properties are just passed to the createElement function, so there is no need for some parallel mechanism to transfer the properties to the component itself.

In single-spa-angular elements the creation of the component just assigns the bare template to the innerHTML and the properties are moved through a global subject - that is the recommended way in the help [for example](https://single-spa.js.org/docs/ecosystem-angular/#custom-props) - but this works only if there is just one instance of the app/element using this subject.

Adding all the props to the template is probably a better way of doing it, but there is a problem with this - it will not work for all types of properties or at least I don't know how to do it. So that's why I have just used the idea from [this comment](https://github.com/single-spa/single-spa-angular/issues/234#issuecomment-894217003) and created a map of subscribers, which is indeed a very hacky solution... 

If you have another idea how to move the passing of the properties inside the creation of the element in the DOM, I'll be glad to implement it...

Reply:
> The problem in fact is the way single-spa-angular is implemented - generally using global subject for the props and passing them in parallel to the creation of the components.

Thanks for this explanation, it helped me understand the issue. Instead of using a singleton Subject to pass the props, the props should just be passed through as attributes/properties to the custom element. I wasn't involved in the original implementation of single-spa-angular's angular elements support, so i don't know what it currently does and am fine changing it. But creating a whole props service when we could just use normal properties seems overkill. Custom elements support properties of any data type (only attributes must be strings). So my first thought is that the data should just be passed to the Angular element as a javascript property rather than an HTML attribute.